torch/csrc/distributed/autograd/utils.cpp

```diff
@@ -47,10 +47,7 @@ DistAutogradContext* addRecvRpcBackward(
   DistAutogradContext& autogradContext =
       autogradContainer.getOrCreateContext(autogradMetadata.autogradContextId);
 
-  if (!tensors.empty()) {
-    TORCH_INTERNAL_ASSERT(
-        torch::autograd::compute_requires_grad(tensors),
-        "Received tensors do not require grad, addRecvRpcBackward should not be called");
+  if (!tensors.empty() && torch::autograd::compute_requires_grad(tensors)) {
```

Comment:
Were the changes here and in python_functions.cpp bugs that were discovered by the unit test in this PR?

Reply:
Yes, I hit errors when running unit test in this PR, but it's kind of expected as we already have #28819 to track this.